pounders/py/clean_up_before_return.py

```diff
@@ -0,0 +1,26 @@
+# This function is called to cleanup X and F, set the exit value, and display
+# reason for exiting
+
+# flag    [dbl] Termination criteria flag:
+#               = 0 normal termination because of grad,
+#               > 0 exceeded nfmax evals,   flag = norm of grad at final X
+#               = -1 if input was fatally incorrect (error message shown)
+#               = -2 model failure
+#               = -3 error from TRSP Solver or if a NaN was encountered
+#               = -4 error in TRSP Solver
+
+
+def clean_up_before_return(X, F, nf, exit_flag):
+    X = X[: nf + 1]
+    F = F[: nf + 1]
+
```

Comment:
The python code below was cleaning up with `nf` and `nf+1`.  MATLAB was cleaning up consistently.  Is this forcing python to obey a single clean-up rule or a bug fix?

Reply:
There was a bug (or at least unexpected behavior) before! 

In `.py` we weren't returning the last entries of `X` and `F` when `F` had a NaN. This further highlights the need for such a single location for cleaning up.